messages/teleporter/message_manager.go

```diff
@@ -105,6 +108,19 @@ func (m *messageManager) ShouldSendMessage(warpMessageInfo *vmtypes.WarpMessageI
 	if !ok {
 		return false, fmt.Errorf("relayer not configured to deliver to destination. destinationChainID=%s", destinationChainID.String())
 	}
+
+	// If allowedDestinations is empty, then all destinations are allowed
+	// If allowedDestinations is not empty, then only the allowed destinations are allowed
+	if _, exist := m.allowedDestinations[destinationChainID]; !exist && len(m.allowedDestinations) > 0 {
```

Comment:
The map returned by `Config.GetSourceIDs` is keyed by `sourceChainID`, not `destinationChainID`.

Reply:
I think the key should be `destinationChainID`. The map return from `Config.GetSouceIDs` is `map[id.ID]map[id.ID]bool`. The first key is sourceChainID, and the second key is destinationChainID. When call [runRelayer](https://github.com/ava-labs/awm-relayer/pull/59/files#diff-327181d0a8c5e6b164561d7910f4eeffd41442d55b2a2788fda2aa2692f17ec0R180), we pass in `map[sourceChainID]`. So here the key of the map is `destinationChainID`.

Reply:
Ah, I did not realize that this is the nested map.